src/charm.py

```diff
@@ -204,6 +236,92 @@ def _reload_pgbouncer(self) -> None:
         pgb_container.restart(PGB)
         self.unit.status = ActiveStatus("PgBouncer Reloaded")
 
+    def add_user(
+        self,
+        user: str,
+        cfg: PgbConfig,
+        password: str = None,
+        admin: bool = False,
+        stats: bool = False,
+        reload_pgbouncer: bool = False,
+        render_cfg: bool = False,
+    ):
+        """Adds a user.
+
+        Users are added to userlist.txt and pgbouncer.ini config files
+
+        Args:
+            user: the username for the intended user
+            cfg: A PgbConfig object. Modified during this method.
+            password: intended password for the user
+            admin: whether or not the user has admin permissions
+            stats: whether or not the user has stats permissions
+            reload_pgbouncer: whether or not to restart pgbouncer after changing config. Must be
+                restarted for changes to take effect.
+            render_cfg: whether or not to render config
+        Returns False if user cannot be created.
```

Comment:
Maybe it's better to return an exception to handle in the method that calls this method? I'm thinking about that because there is no return value for success and also there is no value being returned in some cases (like when the user is already created in the files).

Reply:
You're right, I'll add a new exception in. The charm.py files are maybe getting a little big, but I'm planning to do some refactors once these relations are done.

Reply:
I've actually updated this to use a FileNotFoundError based on our mattermost chat.  There's not a great place to put a dedicated exception with the charm as it exists now, but there's going to be some refactors in the future once the relations are sorted and the charm is released, so I'll add your suggestion then.